runtime/core/exec_aten/util/test/tensor_util_test.cpp

```diff
@@ -605,3 +610,36 @@ TEST_F(TensorUtilTest, SameShapesDifferentDimOrder) {
   EXPECT_FALSE(tensors_have_same_dim_order(a, c, b));
   EXPECT_FALSE(tensors_have_same_dim_order(c, b, a));
 }
+
+TEST_F(TensorUtilTest, TensorShapeToCStringBasic) {
+  std::array<executorch::aten::SizesType, 3> sizes = {123, 456, 789};
+  auto str = tensor_shape_to_c_string(sizes);
+  EXPECT_STREQ(str.data(), "(123, 456, 789)");
+
+  std::array<executorch::aten::SizesType, 1> one_size = {1234567890};
+  str = tensor_shape_to_c_string(one_size);
+  EXPECT_STREQ(str.data(), "(1234567890)");
+}
+
+TEST_F(TensorUtilTest, TensorShapeToCStringMaximumLength) {
+  std::array<
+      executorch::aten::SizesType,
+      executorch::runtime::kTensorDimensionLimit>
+      sizes;
+  std::fill(
+      sizes.begin(),
+      sizes.end(),
+      executorch::runtime::kMaximumPrintableTensorShapeElement);
+
+  auto str = executorch::runtime::tensor_shape_to_c_string(sizes);
+
+  std::ostringstream expected;
+  expected << '(' << kMaximumPrintableTensorShapeElement;
+  for (int ii = 0; ii < kTensorDimensionLimit - 1; ++ii) {
+    expected << ", " << kMaximumPrintableTensorShapeElement;
+  }
+  expected << ')';
+  auto expected_str = expected.str();
+
+  EXPECT_EQ(expected_str, str.data());
+}
```

Comment:
Please also add tests for:
- The "ERR" cases (negative and too large)
- A very-high-dimensional tensor that causes the string to be truncated